### ISSUE TITLE
.NET 6 - Angular: ng serve not running until requesting page

### ISSUE THREAD

**Issue by jepperaskdk** [dotnet/aspnetcore#39314]
When I run `dotnet watch run`, it doesn't run the ng-serve command until I access https://localhost:5001 in my browser.
With .NET 5 it used to start immediately. I have `launchBrowser: false` in my `launchSettings.json`.

.NET: 6.0.101
Angular: 12.2.2

(Sorry, I don't know if this is a design-proposal).

**Comment by hoseinprg** [comment]
> When I run `dotnet watch run`, it doesn't run the ng-serve command until I access https://localhost:5001 in my browser. With .NET 5 it used to start immediately. I have `launchBrowser: false` in my `launchSettings.json`.
> 
> .NET: 6.0.101 Angular: 12.2.2
> 
> (Sorry, I don't know if this is a design-proposal).

I'm not sure, but it might solve the problem:
Assuming you are running the app as a console application (Not using IIS Express) and the angular app resides inside ClientApp folder, you can check the following ways.

If you want to run the angular server independently add this to your Configure() method inside Startup.cs :
```
app.UseSpa(spa =>
        {

            spa.Options.SourcePath = "ClientApp";

            if (env.IsDevelopment())
            {
                spa.UseProxyToSpaDevelopmentServer("http://localhost:4200"); // Use this instead to use the angular cli server
            }
        });
```

If you want to start both of the inside the aspnet core app add this to your Configure() method inside Startup.cs :
```
app.UseSpa(spa =>
        {
            spa.Options.SourcePath = "ClientApp";

            if (env.IsDevelopment())
            {
                spa.UseAngularCliServer(npmScript: "start");
                spa.Options.StartupTimeout = TimeSpan.FromSeconds(60); // Increase the timeout if angular app is taking longer to startup

            }
        });
```

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. @javiercn can you please take a look at this? Thanks!

**Comment by jepperaskdk** [comment]
> > When I run `dotnet watch run`, it doesn't run the ng-serve command until I access https://localhost:5001 in my browser. With .NET 5 it used to start immediately. I have `launchBrowser: false` in my `launchSettings.json`.
> > .NET: 6.0.101 Angular: 12.2.2
> > (Sorry, I don't know if this is a design-proposal).
> 
> I'm not sure, but it might solve the problem: Assuming you are running the app as a console application (Not using IIS Express) and the angular app resides inside ClientApp folder, you can check the following ways.
> 
> If you want to run the angular server independently add this to your Configure() method inside Startup.cs :
> 
> ```cs
> app.UseSpa(spa =>
>         {
> 
>             spa.Options.SourcePath = "ClientApp";
> 
>             if (env.IsDevelopment())
>             {
>                 spa.UseProxyToSpaDevelopmentServer("http://localhost:4200"); // Use this instead to use the angular cli server
>             }
>         });
> ```
> 
> If you want to start both of the inside the aspnet core app add this to your Configure() method inside Startup.cs :
> 
> ```cs
> app.UseSpa(spa =>
>         {
>             spa.Options.SourcePath = "ClientApp";
> 
>             if (env.IsDevelopment())
>             {
>                 spa.UseAngularCliServer(npmScript: "start");
>                 spa.Options.StartupTimeout = TimeSpan.FromSeconds(60); // Increase the timeout if angular app is taking longer to startup
> 
>             }
>         });
> ```

Did you copy the code from here? It looks like .NET 5 or lower.
https://stackoverflow.com/questions/60559070/error-while-running-asp-net-core-with-angular-template

With .NET 6, there seems to no longer be a Startup.cs and the SPA is configured in .csproj, with these values:
```
        <SpaRoot>ClientApp\</SpaRoot>
        <SpaProxyServerUrl>https://localhost:44495</SpaProxyServerUrl>
        <SpaProxyLaunchCommand>npm start</SpaProxyLaunchCommand>
```
Perhaps it can still be configured as you propose,

**Comment by jepperaskdk** [comment]
Any news on this? If I have to restart my server, I have to change port in the browser, to trigger restarting the call to `npm run start`.

**Comment by javiercn** [comment]
We did change this in 6.0 and we only start the proxy on first request. It is normally not an issue since the most common case is to open a page to the app, which triggers the launch. I have however filed an issue [here](https://github.com/dotnet/aspnetcore/issues/44638) to track the work needed to provide the option to start the proxy immediately instead of on first request.

I suspect the reason that we did it this way is because it made some async code to check if the server was already running easier to author.

We do not have plans to change it ourselves unless we receive significant feedback about it, but we would accept PRs as long as they roughly align to what is described in the referenced issue.

I am going to close this issue in favor of the other issue as it has all the context there.
